lib/sample_loader.rb

```diff
@@ -307,12 +307,8 @@ def samples_dir
       unless defined? @samples_dir
         @samples_dir = nil
         base_dir = "#{File.dirname File.dirname __dir__}/"
-        cur_dir = Dir.getwd
-        if cur_dir.start_with? base_dir
-          gem_name = cur_dir[base_dir.length..].split("/").first
-          tentative_dir = File.join base_dir, gem_name, "samples"
-          @samples_dir = tentative_dir if ::File.directory? tentative_dir
-        end
+        tentative_dir = File.join base_dir, "samples"
```

Comment:
I think this still doesn't work because `base_dir` is still the directory within the `ruby-common-tools` repo clone. I suspect what we need to do is start with the current directory (`Dir.curwd`) and walk up the directory tree looking for a gemspec, and then look for a samples directory in the same directory as the gemspec.